File: static/parse.py

def parse(name):
	print(name)
	txt = []
	with open(name) as f:
		# with open('static/cube.obj') as f:
		data = f.read()
	
	data = data.split('\n')
	for i, x in enumerate(data):
		line = x.split(' ')
		if line[0] == 'v':
			txt.append({
			 'type':
			 'v',
			 'coords': [
			  round(float(line[1]) * 10, 5),
			  round(float(line[2]) * 10, 5),
			  round(float(line[3]) * 10, 5)
			 ]
			})
		elif line[0] == 'f':
			index = line[1::]
			index = list(map(lambda a: a.split('/'), index))
			print(index)
			points = list(map(lambda a: str(int(a[0]) - 1), index))
			txt.append({'type': 'f', 'indices': points})
	return txt
# txt = f'export let data = {txt}'

# print(txt)
# with open('static/obj.mjs', 'w') as f:
# 	f.write(txt)

File: static/parse.mjs
import {AddVertex, AddEdge, AddFace} from './main.mjs'
import {data} from './obj.mjs'

export function read() {
	data.forEach(x => {
		switch (x.type) {
			case 'v':
				AddVertex(x.coords[0], x.coords[1], x.coords[2])
				break;
			case 'f':
				// -- works for any polygon --
				// let pairs = []
				// for (let p=0; p<x.indices.length; p++) {
				// 	if (p == x.indices.length-1) {
				// 			pairs.push([x.indices[p], x.indices[0]])
				// 	} else {
				// 			pairs.push([x.indices[p], x.indices[p+1]])
				// 	}
				// }
				// pairs.forEach(p => {AddEdge(p[0], p[1])})

				// -- assumes triangle --
				let points = x.indices
				AddEdge(points[0], points[1])
				AddEdge(points[1], points[2])
				AddEdge(points[2], points[0])
				AddFace(...points)
				break;
		}
	})
}